cpp/net/connection_pool.cc

```diff
@@ -29,21 +145,59 @@ const HostPortPair& ConnectionPool::Connection::other_end() const {
 }
 
 
+void ConnectionPool::Connection::ReleaseConnection() {
+  conn_.release();
+  // Do not null out ssl_, despite the fact that it's a dangling pointer now we
+  // need it in the d'tor since the value of the pointer itself is a key into a
+  // map which needs to be updated.
```

Comment:
I wasn't thinking straight... I meant to have the removal from the map centralized somewhere, like there, and possibly calling that from the destructor, but if it's released, then it won't be freed, and... ugh, you know what I mean. :stuck_out_tongue: